crates/ibc/src/core/ics02_client/handler/create_client.rs

```diff
@@ -28,6 +34,84 @@ pub struct Result {
     pub processed_height: Height,
 }
 
+pub(crate) fn validate<Ctx>(ctx: &Ctx, msg: MsgCreateClient) -> Result<(), Error>
+where
+    Ctx: ValidationContext,
+{
+    let MsgCreateClient {
+        client_state,
+        consensus_state: _,
+        signer: _,
+    } = msg;
+
+    // Construct this client's identifier
+    let id_counter = ctx.client_counter()?;
+
+    let client_state = ctx.decode_client_state(client_state)?;
+
+    let client_type = client_state.client_type();
+
+    let _client_id = ClientId::new(client_type, id_counter).map_err(|e| {
+        Error::client_identifier_constructor(client_state.client_type(), id_counter, e)
+    })?;
+
+    Ok(())
+}
+
+pub(crate) fn execute<Ctx>(ctx: &mut Ctx, msg: MsgCreateClient) -> Result<(), Error>
+where
+    Ctx: ExecutionContext,
+{
+    let MsgCreateClient {
+        client_state,
+        consensus_state,
+        signer: _,
+    } = msg;
+
+    // Construct this client's identifier
+    let id_counter = ctx.client_counter()?;
+
+    let client_state = ctx.decode_client_state(client_state)?;
+
+    let client_type = client_state.client_type();
+
+    let client_id = ClientId::new(client_type.clone(), id_counter).map_err(|e| {
+        Error::client_identifier_constructor(client_state.client_type(), id_counter, e)
+    })?;
+    let consensus_state = client_state.initialise(consensus_state)?;
+
+    ctx.store_client_type(ClientTypePath(client_id.clone()), client_type.clone())?;
```

Comment:
Would be nice if we could extract blocks like these (e.g. `write_create_client()`, `emit_create_client_event()`, etc.) and reused them in the future to avoid code duplication.